modules/quantcastIdSystem.js

```diff
@@ -7,11 +7,157 @@
 
 import {submodule} from '../src/hook.js'
 import { getStorageManager } from '../src/storageManager.js';
+import { triggerPixel, logInfo } from '../src/utils.js';
+import { uspDataHandler, coppaDataHandler, gdprDataHandler } from '../src/adapterManager.js';
 
 const QUANTCAST_FPA = '__qca';
+const DEFAULT_COOKIE_EXP_DAYS = 392; // (13 months - 2 days)
+const PREBID_PCODE = 'p-KceJUEvXN48CE';
+const QSERVE_URL = '//pixel.quantserve.com/pixel';
+const QUANTCAST_VENDOR_ID = '11';
+const PURPOSE_DATA_COLLECT = '1';
+const PURPOSE_PRODUCT_IMPROVEMENT = '10';
+const QC_TCF_REQUIRED_PURPOSES = [PURPOSE_DATA_COLLECT, PURPOSE_PRODUCT_IMPROVEMENT];
+const QC_TCF_CONSENT_FIRST_PURPOSES = [PURPOSE_DATA_COLLECT];
+const QC_TCF_CONSENT_ONLY_PUPROSES = [PURPOSE_DATA_COLLECT];
+const GDPR_PRIVACY_STRING = gdprDataHandler.getConsentData();
+const US_PRIVACY_STRING = uspDataHandler.getConsentData();
 
 export const storage = getStorageManager();
 
+export function firePixel(clientId, cookieExpDays = DEFAULT_COOKIE_EXP_DAYS) {
+  // check for presence of Quantcast Measure tag _qevent obj and publisher provided clientID
+  if (!window._qevents && clientId && clientId != '') {
+    var fpa = storage.getCookie(QUANTCAST_FPA);
+    var fpan = '0';
+    var domain = quantcastIdSubmodule.findRootDomain();
+    var now = new Date();
+    var usPrivacyParamString = '';
+    var firstPartyParamStrings;
+    var gdprParamStrings;
+
+    if (!fpa) {
+      var expires = new Date(now.getTime() + (cookieExpDays * 86400000)).toGMTString();
```

Comment:
Seems weird to call `now.getTime()` twice in a row like this. Also weird to use string interpolation below and not here.

Why not do this instead?

```
const DAY_MS = 86400000;
/* ... *snip* ... */
```

```
var et = now.getTime();
var expires = new Date(et + (cookieExpDays * DAY_MS)).toGMTString();
var rand = Math.round(Math.random() * 2147483647);
fpa = "B0-${rand}-${et}";
```

Reply:
Otherwise :+1: